main/config.c

```diff
@@ -255,17 +255,38 @@ struct ast_config {
 	struct ast_category *last_browse;     /*!< used to cache the last category supplied via category_browse */
 	int include_level;
 	int max_include_level;
-	struct ast_config_include *includes;  /*!< a list of inclusions, which should describe the entire tree */
+	/*! First inclusion in the list. */
+	struct ast_config_include *includes_root;
+	/*! Last inclusion in the list. */
+	struct ast_config_include *includes_last;
 };
 
+struct config_included_file {
+	AST_LIST_ENTRY(config_included_file) next;
+	/*|< #include path or matching #tryinclude paths */
+	char path[];
+};
+
+/*!
+ * \brief Types used for ast_include_new() include_type
+ */
+typedef enum {
```

Comment:
Do not make a typedef for this, just use this as an enum directly: https://docs.asterisk.org/Development/Policies-and-Procedures/Coding-Guidelines/#dont-use-unnecessary-typedefs

Reply:
Fixed